### PR TITLE
Prep for 1.1.2 release with cleaned up DEBUG flag logging

### PR DESCRIPTION
**Pull request by dhagberg-sf**
Related to
* W-7379871
* https://gus.my.salesforce.com/a07B0000008BAB9IAO

Unclear log output w/the word "undefined"
Use sdk 1.1.2 w/lowered log level

Prep for release 1.1.2.

### CODE REVIEW

File: middleware/index.ts [hunk 1]
```diff
@@ -221,14 +217,16 @@ export default function applySfFxMiddleware(request: any, state: any, resultArgs
 
     //use secret here in lieu of DEBUG runtime environment var until we have deployment time support of config var
     const debugSecret = getSecret("sf-debug", "DEBUG", logger);
-    logger.info(`DEBUG flag is ${debugSecret}`);
     if(debugSecret || LoggerLevel.DEBUG === logger.getLevel() || process.env.DEBUG) {
+        logger.info(`DEBUG flag is ${debugSecret}`);
```
Comment (jqian2017) [inline middleware/index.ts]: @dhagberg-sf , this needs to be outside of the if, if it's already inside if, I already know it's set

the following is what I have

    `logger.info(`DEBUG flag is ${debugSecret ? debugSecret : 'unset'}`);`

File: middleware/index.ts [hunk 2]
```diff
@@ -190,11 +190,7 @@ function createContext(id: string, logger: Logger, reqContext?: any, accessToken
  * @return secret value if successfully loaded, undefined if not found/loaded.
  */
 function getSecret(name: string, key: string, logger: Logger) : string|undefined {
-    const secretMap = new Secrets(logger).get(name);
-    if (secretMap && key in secretMap) {
-        return secretMap[key];
-    }
-    return undefined;
+    return new Secrets(logger).getValue(name, key);
```
Comment (jqian2017) [inline middleware/index.ts]: @dhagberg-sf Should we use the same Secret for both retrieving the sf-debug secret as well as using it to init the context later?

File: middleware/index.ts [hunk 3]
```diff
@@ -221,14 +217,16 @@ export default function applySfFxMiddleware(request: any, state: any, resultArgs
...
     if(debugSecret || LoggerLevel.DEBUG === logger.getLevel() || process.env.DEBUG) {
+        logger.info(`DEBUG flag is ${debugSecret}`);
         //for dev preview, we log the ENTIRE raw request, may need to filter sensitive properties out later
         //the hard part of filtering is to know which property name to filter
         //change the logger level, so any subsequent user function's logger.debug would log as well
         logger.setLevel(LoggerLevel.DEBUG);
         logger.debug('debug raw request in middleware');
         logger.debug(request);
+    } else {
+        logger.info('DEBUG flag is off');
```
Comment (cwallsfdc) [inline middleware/index.ts]: If the log lines in the true block are omitted, then we know debug is not set.  I suppose I'm not sure why we need to state that debug is disabled.

File: middleware/index.ts [hunk 4]
```diff
@@ -158,18 +158,21 @@ function createOrg(logger: Logger, reqContext: any, accessToken?: string): Org {
...
  */
-function createContext(id: string, logger: Logger, reqContext?: any, accessToken?: string, functionInvocationId?: string): Context {
+function createContext(id: string, logger: Logger, secrets: Secrets, reqContext?: any,
+                       accessToken?: string, functionInvocationId?: string): Context {
     if (typeof reqContext === 'string') {
         reqContext = JSON.parse(reqContext);
     }
 
     const org = reqContext ? createOrg(logger, reqContext!, accessToken) : undefined;
-    const secrets = createSecrets(logger);
```
Comment (dhagberg-sf) [inline middleware/index.ts]: Moved to initial section of `applySfFxMiddleware` to ensure only one Secrets created at invoke time

File: middleware/index.ts [hunk 5]
```diff
@@ -181,22 +184,6 @@ function createContext(id: string, logger: Logger, reqContext?: any, accessToken
...
 
-/**
- * Load a single secret/key to check at startup.
- *
- * @param name -- secret name
- * @param key -- key within secret to check
- * @param logger -- to use to log secret load errors
- * @return secret value if successfully loaded, undefined if not found/loaded.
- */
-function getSecret(name: string, key: string, logger: Logger) : string|undefined {
```
Comment (dhagberg-sf) [inline middleware/index.ts]: logic is already in Secrets.getValue now so not needed here

File: middleware/index.ts (hunk 1 above)
Comment (dhagberg-sf) [reply middleware/index.ts]: fixed.

File: middleware/index.ts (hunk 3 above)
Comment (dhagberg-sf) [reply middleware/index.ts]: fixed per slack discussion - for support purposes it's better if we *always* log the DEBUG flag during dev preview - helps us know if we even hit that code section.

File: middleware/index.ts (hunk 2 above)
Comment (dhagberg-sf) [reply middleware/index.ts]: Yes we should reuse same Secrets object - fixed.

**Pull request merged**